### PR TITLE
add jaeger support for codec

### LINKED ISSUE #6389: Enhance jaeger tracing properties to allow propagation
### Feature description

Currently you can add `sender`, `reporter` and `sampler` properties for jaeger tracing, but you cannot add propagation. 
This means that if you want your micronaut service to take part in a distributed trace with another service using the default opentelemetry javaagent, you have to set an environment variable of `JAEGER_PROPAGATION: W3C` 
It would be good if the `io.micronaut.tracing.jaeger.JaegerConfiguration` class could be modified to take `CodecConfiguration` as well, something like:
```
tracing:
  jaeger:
    enabled: true
    codec:
       codecs: W3C
```

### PR DESCRIPTION
**Pull request by manuelmartinoracle**
Fixes https://github.com/micronaut-projects/micronaut-core/issues/6389

### CODE REVIEW

**Comment by graemerocher** [comment]
Please correct the checkstyle errors

File: test-suite/src/test/groovy/io/micronaut/tracing/jaeger/JaegerConfigurationSpec.groovy [hunk 1]
```diff
@@ -22,4 +29,176 @@ class JaegerConfigurationSpec extends Specification {
         cleanup:
         ctx.close()
     }
+
+    /**
```
Comment (pgressa) [inline test-suite/src/test/groovy/io/micronaut/tracing/jaeger/JaegerConfigurationSpec.groovy]: we don't add author annotation for tests

File: test-suite/src/test/groovy/io/micronaut/tracing/jaeger/JaegerConfigurationSpec.groovy [hunk 2]
```diff
@@ -22,4 +29,176 @@ class JaegerConfigurationSpec extends Specification {
...
+        headerCodecs.get(1) instanceof B3TextMapCodec // B3
+        headerCodecs.get(2) instanceof TextMapCodec // JAEGER
+
+        textMapCodecs.get(0) instanceof TraceContextCodec // W3C
+        textMapCodecs.get(1) instanceof B3TextMapCodec // B3
+        textMapCodecs.get(2) instanceof TextMapCodec // JAEGER
+
+        jaegerCodecs.get(0) instanceof BinaryCodec // JAEGER
+
+        /*
```
Comment (pgressa) [inline test-suite/src/test/groovy/io/micronaut/tracing/jaeger/JaegerConfigurationSpec.groovy]: remove commented code

File: tracing/src/main/java/io/micronaut/tracing/jaeger/JaegerConfiguration.java [hunk 3]
```diff
@@ -182,6 +182,18 @@ public void setCodecConfiguration(@Nullable Configuration.CodecConfiguration cod
         }
     }
 
+    /**
+     * Sets the codec configuration.
+     * @author Manuel Martin
```
Comment (pgressa) [inline tracing/src/main/java/io/micronaut/tracing/jaeger/JaegerConfiguration.java]: we don't add author annotation to methods, move it to the class javadoc

File: tracing/src/main/java/io/micronaut/tracing/jaeger/JaegerConfiguration.java [hunk 4]
```diff
@@ -277,4 +289,56 @@ public void setSenderConfiguration(@Nullable JaegerSenderConfiguration senderCon
             return configuration;
         }
     }
+
+    /**
+     * The codec configuration bean
+     * @author Manuel Martin
+     * @author Burt Beckwith
+     */
+    @ConfigurationProperties("codec")
```
Comment (pgressa) [inline tracing/src/main/java/io/micronaut/tracing/jaeger/JaegerConfiguration.java]: I don't think it's necessary to add a new configuration property class that encapsulates just one string field. Is there a problem to have the `codec` (`codecs`) to be part of `JaegerConfiguration` ?

File: test-suite/src/test/groovy/io/micronaut/tracing/jaeger/JaegerConfigurationSpec.groovy [hunk 5]
```diff
@@ -22,4 +29,176 @@ class JaegerConfigurationSpec extends Specification {
...
+     * @author Burt Beckwith
+     * @autohr Manuel Martin
+     */
+    void "test codec configuration W3C B3 JAEGER"() {
+        given:
+        def ctx = ApplicationContext.run(
+                'tracing.jaeger.enabled':'true',
+                'tracing.jaeger.codec.codecs':'W3C,B3,JAEGER'
+        )
+        def config = ctx.getBean(JaegerConfiguration.JaegerCodecConfiguration).getCodecConfiguration()
```
Comment (pgressa) [inline test-suite/src/test/groovy/io/micronaut/tracing/jaeger/JaegerConfigurationSpec.groovy]: could the subject of test be the `io.jaegertracing.Configuration` aka `JaegerConfiguration#getConfiguration` ?

**Review by pgressa — CHANGES_REQUESTED** [review]
Good job, please remove commented code and address the code review comments.

**Review by pgressa — CHANGES_REQUESTED** [review]
Good job @manuelmartinoracle ! could you please extend the documentation as well https://docs.micronaut.io/latest/guide/#jaeger,  thank you!

**Review by pgressa — APPROVED** [review]

**Pull request merged**